torch/fx/experimental/symbolic_shapes.py

```diff
@@ -516,13 +514,42 @@ def _constrain_range_for_size(a, min: Optional[int] = None, max: Optional[int] =
             "received min={min} and max={max}"
         )
 
+    shape_env = a.node.shape_env
+
+    vr = ValueRanges(min, max)
+    if free_unbacked_symbols(a.node.expr):
+        if (
+            not isinstance(a.node.expr, sympy.Symbol) and
+            not bound_sympy(a.node.expr, shape_env.var_to_range).issubset(vr)
+        ):
+            # Generate a new unbacked SymInt and setup a replacement.  The new
+            # unbacked SymInt can be marked as size like and it can also
+            # now have a refined range
+            u1 = shape_env.create_unbacked_symint().node.expr
+            # TODO: Also use whatever old range we knew from the expression
+            shape_env.var_to_range[u1] = vr
+            # Log this before the replacement occurs, so we can get the old
+            # a.node.expr
+            log.info("_constrain_range_for_size %s = %s [%s, %s]", a.node.expr, u1, min, max)
+            shape_env.complex_replacements[a.node.expr] = u1
+            shape_env._update_version_counter()
+            assert isinstance(a.node.expr, sympy.Symbol)
+    else:
+        assert isinstance(a.node.expr, sympy.Symbol), "constraining on non-symbol backed SymInt NYI"
+        # TODO: Could we do the unbacked replacement strategy here?  It's
```

Comment:
Replace a backed symbol by an unbacked one? Why would we want that?

Reply:
Replace backed with backed. And for the same reason: you can reliably refine the range afterwards.